Makefile.one

```diff
@@ -0,0 +1,443 @@
+	@echo "$ make non-core             # build ./dist/tracee.bpf.XXX.o"
+	@echo "$ make tracee-ebpf          # build ./dist/tracee-ebpf"
+	@echo "$ make tracee-rules         # build ./dist/tracee-rules"
+	@echo "$ make rules                # build ./dist/rules"
+	@echo ""
+	@echo "# clean"
+	@echo ""
+	@echo "$ make clean                # wipe ./dist/"
+	@echo "$ make clean-core           # wipe ./dist/tracee.bpf.core.o"
+	@echo "$ make clean-non-core       # wipe ./dist/tracee.bpf.XXX.o"
+	@echo "$ make clean-tracee-ebpf    # wipe ./dist/tracee-ebpf"
+	@echo "$ make clean-tracee-rules   # wipe ./dist/tracee-rules"
+	@echo "$ make clean-rules          # wipe ./dist/rules"
+	@echo ""
+	@echo "# test"
+	@echo ""
+	@echo "$ make test                 # run all go & opa tests"
+	@echo "$ make test-tracee-ebpf     # go test tracee-ebpf"
+	@echo "$ make test-tracee-rules    # go test tracee-rules"
+	@echo "$ make test-rules           # opa test (tracee-rules)"
+	@echo ""
+
+# variables
+
+BPF_VCPU = v2
+
+# requirements
+
+.PHONY: reqs_cmds
+reqs_cmds: \
+	check_$(CMD_GIT) \
+	check_$(CMD_CLANG) \
+	check_$(CMD_LLC) \
+	check_$(CMD_STRIP) \
+	check_$(CMD_RM) \
+	check_$(CMD_GREP) \
+	check_$(CMD_INSTALL) \
+	check_$(CMD_MKDIR) \
+	check_$(CMD_PKGCONFIG) \
+	check_$(CMD_GO) \
+	check_$(CMD_OPA)
+
+.PHONY: reqs_libs
+reqs_libs: \
+	checklib_$(LIB_ELF) \
+	checklib_$(LIB_ZLIB)
+
+.PHONY: reqs
+reqs:
+	$(MAKE) reqs_cmds
+	$(MAKE) reqs_libs
+
+# output dir
+
+OUTPUT_DIR = ./dist
+
+$(OUTPUT_DIR):
+	$(CMD_MKDIR) -p $@
+	$(CMD_MKDIR) -p $@/libbpf
+	$(CMD_MKDIR) -p $@/libbpf/obj
+
+# bundle
+
+$(OUTPUT_DIR)/tracee.bpf: $(OUTPUT_DIR)/libbpf/libbpf.a
+	$(CMD_MKDIR) -p $@
+	$(CMD_INSTALL) -m 0640 ./tracee-ebpf/3rdparty/include/* $@
+	$(CMD_INSTALL) -m 0640 $(OUTPUT_DIR)/libbpf/bpf/*.h $@
+	$(CMD_INSTALL) -m 0640 $(TRACEE_EBPF_OBJ_SRC) $@
+
+#
+# libbpf
+#
+
+LIBBPF_CFLAGS =
+LIBBPF_LDLAGS =
+LIBBPF_SRC = ./tracee-ebpf/3rdparty/libbpf/src
+
+$(OUTPUT_DIR)/libbpf/libbpf.a: $(LIBBPF_SRC) $(wildcard $(LIBBPF_SRC)/*.[ch]) | $(OUTPUT_DIR)
+	CC="$(CMD_CLANG)" \
+		CFLAGS="$(LIBBPF_CFLAGS)" \
+		LD_FLAGS="$(LIBBPF_LDFLAGS)" \
+		$(NORM_MAKE) -C $(LIBBPF_SRC) \
+		BUILD_STATIC_ONLY=1 \
+		DESTDIR=$(abspath ./$(OUTPUT_DIR)/libbpf/) \
+		OBJDIR=$(abspath ./$(OUTPUT_DIR)/libbpf/obj) \
+		INCLUDEDIR= LIBDIR= UAPIDIR= prefix= libdir= \
+		install install_uapi_headers
+
+$(LIBBPF_SRC):
+ifeq ($(wildcard $@), )
+	$(CMD_GIT) submodule update --init --recursive
+endif
+
+#
+# non co-re ebpf
+#
+
+KERN_RELEASE ?= $(UNAME_R)
+KERN_BUILD_PATH ?= $(if $(KERN_HEADERS),$(KERN_HEADERS),/lib/modules/$(KERN_RELEASE)/build)
+KERN_SRC_PATH ?= $(if $(KERN_HEADERS),$(KERN_HEADERS),$(if $(wildcard /lib/modules/$(KERN_RELEASE)/source),/lib/modules/$(KERN_RELEASE)/source,$(KERN_BUILD_PATH)))
+
+NONCORE_TAG = $(subst .,_,$(KERN_RELEASE)).$(subst .,_,$(VERSION))
+
+.PHONY: non-core
+non-core: $(OUTPUT_DIR)/tracee.bpf.$(NONCORE_TAG).o
```

Comment:
can we give this target a more meaningfull name like `bpf-nocore`

Reply:
bpf-nocore